content/docs/3_cookbook/0_development-deployment/0_caddy/cookbook-recipe.txt

```diff
@@ -77,37 +82,41 @@ We're creating two custom **snippets** here with the names `common` and `kirby`.
 The `php_fastcgi` directive also comes with a lot of good default values for PHP-specific configuration values (e.g. it automatically sets `try_files` to look for an `index.php` file – so a request gets forwarded to PHP when there is no file with a specific name)
 <br/>
 
+
 ```
-tls name@user.com
+file_server
 ```
-
-This directive will take care of your TLS certificate. If you want to use a free certificate from Let's Encrypt, enter your email address here and you're done. If you want to run your Kirby site locally, use `tls internal` for a self-signed certificate.
+This directive will make sure Caddy serves all static assets, e.g. all of the JavaScript, font files, icons, images, etc. Without this directive, only PHP files would be served/interpreted.
 <br/>
 
-
 ```
-file_server
+encode zstd gzip
 ```
-
-This directive will make sure Caddy serves all static assets, e.g. all of the JavaScript, font files, icons, images, etc. Without this directive, only PHP files would be served/interpreted.
+We'll enable compression to reduce the transfered file size. This will make your site faster and has been supported by all major browsers for many years.
 <br/>
 
 
 ```
 @blocked {
-	path *.txt *.md /content/* /site/* /kirby/* /.*
+	path /content/* /site/* /kirby/* /.*
 }
-redir @blocked /
+error @blocked "Not found" 404
 ```
-
-Here we are defining a custom matcher – which is a statement that we can then use in multiple locations as an argument for other directives. Our matcher is called `@blocked` – because we're defining paths which should not be accessible by website visitors: all raw txt/md files and everything in the `content`, `site` or `kirby` folders, as well as hidden files. In the following line, we're redirecting all these requests to the home page of our site.
+Here we are doing some Kirby-specific error handling. First, we're defining a custom matcher – which is a statement that we can then use in multiple locations as an argument for other directives. Our matcher is called `@blocked` – because we're defining paths which should not be accessible by website visitors: everything in the `content`, `site` or `kirby` folders, as well as hidden files. In the following line, we're returning a 404 HTTP error, when any of these files are being requested.
 <br/>
 
+```
+handle_errors {
+	rewrite * /error # This should reference your Kirby error page
+	import common
+}
+```
+The error we're returning from the line before will result in the browser showing an empty page – which is typically not what we want. Instead, we want to show Kirbys error page, which is served under the `/error` path by default. So we're rewriting all the requests coming into this error handler to the Kirby error page. And we also need the `common` block here, so PHP will be executed as well (because Kirbys error page is not a static HTML file, but is dynamically generated via PHP).
```

Comment:
```suggestion
The error we're returning from the line before will result in the browser showing an empty page – which is typically not what we want. Instead, we want to show Kirby's error page, which is served under the `/error` path by default. So we're rewriting all the requests coming into this error handler to the Kirby error page. And we also need the `common` block here, so PHP will be executed as well (because Kirby's error page is not a static HTML file, but is dynamically generated via PHP).
```